perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -1539,7 +1539,7 @@ public static LocalDate extendDateByStaticDate(LocalDate localDate, Matcher matc
 		extensionInNextYear = extensionDate.isBefore(LocalDate.now());
 
 		// Set the date to which the membership should be extended, can be changed if there was grace period, see next part of the code
-		localDate = LocalDate.of(year, month, day);
+		LocalDate localDate = LocalDate.of(year, month, day);
```

Comment:
At this moment, you don't need to use localDate, you can use extensionDate and return it.